sysdeps/aero/generic/thread.cpp

```diff
@@ -17,12 +17,17 @@ extern "C" void __mlibc_enter_thread(void *entry, void *user_arg, Tcb *tcb) {
     if (mlibc::sys_tcb_set(tcb))
         __ensure(!"sys_tcb_set() failed");
 
-    void *(*func)(void *) = reinterpret_cast<void *(*)(void *)>(entry);
-    auto result = func(user_arg);
+	if(tcb->returnValueType == TcbThreadReturnValue::ThreadReturnsVoidPtr) {
```

Comment:
Might be nice to create a helper function for this so sysdeps don't have to concern themselves with this particular peculiarity.

Reply:
something that returns an `std::variant<void*,int>` or something along those lines?

Reply:
No i'd rather see an ``mlibc::invoke_thread_func(tcb, user_arg)``, or even a ``Tcb::invoke_thread_func(user_arg)`` function.

Reply:
The result is only used to set the return value field in the TCB.

Reply:
ah, true, I hadn't thought of that. in that case I'd agree adding this as a function to `Tcb` is probably the cleanest, especially because this is basically just boilerplate that every sysdep would repeat